[Art/96894] - poa search tabs

jquispe-oddball opened the pull request:
Adding tabs to the POA Requests page, [see existing ticket ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/96894).

Merged in [existing PR](https://github.com/department-of-veterans-affairs/vets-website/pull/33441) to get tabs to work with loaders.

Some notes:
- pending tab state is the default state on page load
- we are no longer showing a status badge for pending requests


https://github.com/user-attachments/assets/a4b95509-2b84-4991-a910-7bd3497f3a64

chumpy committed:
switching to router loader data fetching approach

chumpy committed:
adding Suspense

jquispe-oddball committed:
96894 - adding tabs to poa request

jquispe-oddball committed:
96894 - reverting update to loader for now

jquispe-oddball committed:
96894 - adding missing border bottom on mobile

nihil2501 commented on src/applications/accredited-representative-portal/containers/POARequestsPage.jsx:
```diff
@@ -1,17 +1,46 @@
+                to="/poa-requests/pending"
+                className="poa-request__tab-link"
+                role="tab"
+                aria-selected={({ isActive }) => (isActive ? 'true' : 'false')}
```
I don't see that this is one of the props that `NavLink` will accept a function for.

jquispe-oddball replied:
you are right - it is not reflecting the correct state. I grabbed this from the react router docs and figured it would work for aria-selected.I _thought_ I also saw this on other files as well, but will double check the implementation

jquispe-oddball replied:
re: the font comment - while working on accept/decline flow ticket, I also see that the fonts are not loading for me.

nihil2501 commented on src/applications/accredited-representative-portal/containers/POARequestsPage.jsx:
```diff
@@ -1,17 +1,46 @@
       <div className="poa-requests-page-table-container">
-        <h2 data-testid="poa-requests-table-heading">Requests</h2>
-        <POARequestsTableFetcher usePOARequests={usePOARequests} />
+        {poaRequests.length === 0 ? (
+          <p data-testid="poa-requests-table-fetcher-no-poa-requests">
+            No POA requests found
+          </p>
```
Tabs should continue exist even when fetching poa requests returns no results.

jquispe-oddball replied:
going forward can we write better tickets that reflect all the different types of views that a rep may see? I know we are doing a lot of things at once, but have noticed little things that are not included in some tickets.

nihil2501 replied:
Yeah, we need better descriptions in tickets. Candi and I will work on refining some of our existing tickets further.

jquispe-oddball replied:
updated so that the pending view is the default view

nihil2501 commented on src/applications/accredited-representative-portal/containers/POARequestsPage.jsx:
```diff
@@ -1,17 +1,46 @@
+                Pending requests
+              </NavLink>
+              <NavLink
+                to="/poa-requests/completed"
```
These tabs contribute to the query that we use to search for POA requests. This query will be reflected in the query string parameters like `/poa-requests?status=completed`.

jquispe-oddball replied:
see https://github.com/department-of-veterans-affairs/vets-website/pull/33602#discussion_r1889340810

nihil2501 commented on src/applications/accredited-representative-portal/containers/POARequestsPage.jsx:
```diff
@@ -1,17 +1,46 @@
+          <>
+            <div role="tablist" className="poa-request__tabs">
+              <NavLink
+                to="/poa-requests/pending"
```
These tabs contribute to the query that we use to search for POA requests. This query will be reflected in the query string parameters like `/poa-requests?status=pending`.

jquispe-oddball replied:
I reached out to kevin about this prior to submitting pr since params were not working for me at all. I left it like this so the tabs would actually work for my video example. We don't have an endpoint for status to test with the loaders right now, just mock data.

nihil2501 commented on src/applications/accredited-representative-portal/routes.jsx:
```diff
@@ -22,10 +25,32 @@ const router = createBrowserRouter(
               element: <POARequestsPage />,
+              loader: poaRequestsLoader,
+              errorElement: <ErrorMessage />,
+              children: [
+                {
+                  index: true,
+                  loader: poaRequestsLoader,
+                  element: <POARequestsCard cssClass="pending" />,
+                },
+                {
+                  index: true,
+                  path: 'pending',
+                  loader: poaRequestsLoader,
+                  element: <POARequestsCard cssClass="pending" />,
+                },
+                {
+                  path: 'completed',
+                  // loader: poaRequestsCompletedLoader,
+                  element: <POARequestsCard cssClass="completed" />,
+                },
+              ],
```
- We still want the previous route structure
- Tab queries go in query parameters
- The Card component is for a single POA request, `POARequestsPage` is for rendering a list of cards

jquispe-oddball replied:
updated

nihil2501 reviewed (dismissed):
- No results doesn't mean that tabs shouldn't render. They ought to be able to try a new query (change tab)
- `NavLink` API `aria-selected`?
- When building off another branch, branch off of it and open a PR into the branch you are branching off of: `branch 1 <- branch 2`

From the video, it seems like there are font regressions in multiple areas?

jquispe-oddball committed:
96894 - reverting from bad merge

jquispe-oddball committed:
96894 - tab update

jquispe-oddball committed:
96894 - tabs

jquispe-oddball committed:
96894 - fix active state on navlink tabs

jquispe-oddball committed:
96894- removing line

jquispe-oddball committed:
96894 - cypress and linter error fix

jquispe-oddball committed:
96894 - revert

nihil2501 committed:
[ART] POARequestsPage -> POARequestSearchPage

nihil2501 committed:
[ART] Fix typos in POARequestsCard unit spec

nihil2501 committed:
[ARF] Consolidate `StatusTabLink`
Query params are not the same sort of thing as path segments in `react-router` and `NavLink` won't understand them in the same way.

nihil2501 committed:
[ART] Inline POA request list in search page

nihil2501 committed:
[ART] re-org POA search components & fix: retain tabs in empty state

nihil2501 committed:
[ART] Redirect POA request search param

nihil2501 committed:
[ART] fix POA request card filename

nihil2501 committed:
[ART] skip cypress tests that can't understand react-router redirect

chumpy committed:
removing loaders

chumpy committed:
Merge branch 'main' into art/96894/poa-search-tabs
